rcl/src/rcl/subscription.c

```diff
@@ -116,8 +119,12 @@ rcl_subscription_init(
   }
   subscription->impl->actual_qos.avoid_ros_namespace_conventions =
     options->qos.avoid_ros_namespace_conventions;
-  // options
-  subscription->impl->options = *options;
+  ret = rcl_subscription_options_copy(options, &subscription->impl->options);
```

Comment:
I think at this time, calling `rnw_create_subscription` is to create `parent` subscription for content filtering. so actually it just saves `filter_expression` and `expression_parameters`, not to be used during creating subscription. am i understanding right? and beside, at this moment, it is likely that `filter_expression` and `expression_parameters` are empty, right? because `rcl_subscription_set_cft_expression_parameters` cannot be called before.

Reply:
> I think at this time, calling `rnw_create_subscription` is to create `parent` subscription for content filtering. so actually it just saves `filter_expression` and `expression_parameters`, not to be used during creating subscription. am i understanding right? 

The concrete rmw(rmw_connextdds) had already directly used `filter_expression` and `expression_parameters`
https://github.com/iuhilnehc-ynos/rcl/blob/6ef26c8133da5e61360db13502be4fef62938723/rcl/src/rcl/subscription.c#L101, 
I thought that these two parameters should be backed up into `rcl_subscription_impl_t`, so it can prevent from accessing a pointer that might be changed outside.

> and beside, at this moment, it is likely that `filter_expression` and `expression_parameters` are empty, right? because `rcl_subscription_set_cft_expression_parameters` cannot be called before.

Yes.